flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/plan/nodes/exec/TestingExecNode.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.nodes.exec
+
+import org.apache.flink.api.dag.Transformation
+import org.apache.flink.table.api.TableConfig
+import org.apache.flink.table.planner.calcite.{FlinkContextImpl, FlinkRelOptClusterFactory, FlinkRexBuilder, FlinkTypeFactory, FlinkTypeSystem}
+import org.apache.flink.table.planner.delegation.BatchPlanner
+import org.apache.flink.table.planner.plan.nodes.physical.batch.BatchPhysicalRel
+
+import org.apache.calcite.plan.hep.{HepPlanner, HepProgram}
+import org.apache.calcite.plan.{Convention, RelOptCluster, RelOptCost, RelOptPlanner, RelOptQuery, RelOptTable, RelTraitSet}
+import org.apache.calcite.rel.`type`.{RelDataType, RelDataTypeSystem}
+import org.apache.calcite.rel.core.CorrelationId
+import org.apache.calcite.rel.{RelCollation, RelNode, RelShuttle, RelVisitor, RelWriter}
+import org.apache.calcite.rel.metadata.{Metadata, RelMetadataQuery}
+import org.apache.calcite.rex.{RexNode, RexShuttle}
+import org.apache.calcite.util.{ImmutableBitSet, Litmus}
+
+import java.util
+
+/**
+ * [[BatchExecNode]] for testing purpose.
+ */
+class TestingExecNode extends BatchExecNode[BatchPlanner] with BatchPhysicalRel {
```

Comment:
rename to `TestingBatchExecNode` ?
also extends from `AbstractRelNode`, which has implemented many methods.